src/NodeCompiler/CompileNodeToValue.php

```diff
@@ -273,6 +281,30 @@ private function compileDirConstant(Node\Scalar\MagicConst\Dir $node, CompilerCo
         return dirname(realpath($context->getFileName()));
     }
 
+    /**
+     * Compiles magic constant __CLASS__
+     *
+     * @param Node\Scalar\MagicConst\Class_ $node
+     * @param CompilerContext $context
+     * @return string
+     */
+    private function compileClassConstant(Node\Scalar\MagicConst\Class_ $node, CompilerContext $context): string
+    {
+        return $context->hasSelf() ? $context->getSelf()->getName() : '';
```

Comment:
This matches PHP's behaviour (https://3v4l.org/aIRNG), but this feels wrong... should we throw an exception if context doesn't `hasSelf()`? It doesn't make sense to use `__CLASS__` outside of a class context...